### PR TITLE
fby3: dl: Add Sensor HSC MP5990

### PR DESCRIPTION
**Pull request by YoungLi123**
Summary:
- Add Sensor HSC MP5990

Test plan:
- Build code: Pass
- Get HSC sensors and Outlet temp correctly

MP5990:
MB Outlet Temp               (0x2) :   37.00 C     | (ok)
HSC Temp                     (0xF) :   37.00 C     | (ok)
HSC Input Vol                (0x26) :   11.64 Volts | (ok)
HSC Output Cur               (0x30) :    2.40 Amps  | (ok)
HSC Input Pwr                (0x2E) :   28.00 Watts | (ok)
HSC Input AvgPwr             (0x39) :   28.00 Watts | (ok)

### CODE REVIEW

File: meta-facebook/yv3-dl/src/platform/plat_sensor_table.c [hunk 1]
```diff
@@ -337,6 +359,25 @@ void load_sensor_config(void)
...
+	uint8_t hsc_module = get_hsc_module();
+	switch (hsc_module) {
+	case HSC_MODULE_MP5990:
+		extend_sensor_config_size += ARRAY_SIZE(mp5990_sensor_config_table);
+		break;
+	case HSC_MODULE_LTC4282:
+		extend_sensor_config_size += ARRAY_SIZE(ltc4282_sensor_config_table);
+		break;
+	default:
+		printf("[%s] unsupported HSC module, HSC module: 0x%x\n", __func__, hsc_module);
```
Comment (GoldenBug) [inline meta-facebook/yv3-dl/src/platform/plat_sensor_table.c:374]: LOG_ERR(...)
Remove newline.
Remove __func__ argument.

File: meta-facebook/yv3-dl/src/platform/plat_sensor_table.c [hunk 2]
```diff
@@ -447,6 +489,24 @@ void pal_extend_sensor_config()
...
+		}
+		break;
+	case HSC_MODULE_LTC4282:
+		sensor_count = ARRAY_SIZE(ltc4282_sensor_config_table);
+		for (int index = 0; index < sensor_count; index++) {
+			add_sensor_config(ltc4282_sensor_config_table[index]);
+		}
+		break;
+	default:
+		printf("[%s] unsupported HSC module, HSC module: 0x%x\n", __func__, hsc_module);
```
Comment (GoldenBug) [inline meta-facebook/yv3-dl/src/platform/plat_sensor_table.c:506]: LOG_ERR(...)
Remove newline.
Remove __func__ argument.

File: common/dev/mp5990.c [hunk 3]
```diff
@@ -24,6 +25,79 @@
...
+	static bool pre_ein = false;
+
+	msg.bus = sensor_config[sensor_config_index_map[sensor_num]].port;
+	msg.target_addr = sensor_config[sensor_config_index_map[sensor_num]].target_addr;
+	msg.tx_len = 1;
+	msg.data[0] = sensor_config[sensor_config_index_map[sensor_num]].offset;
+	msg.rx_len = 7;
+
+	if (i2c_master_read(&msg, retry)) {
+		LOG_WRN("i2c read failed.\n");
```
Comment (GoldenBug) [inline common/dev/mp5990.c:55]: remove newline

File: common/dev/mp5990.c [hunk 4]
```diff
@@ -24,6 +25,79 @@
...
+	}
+
+	if (pre_sample > sample) {
+		sample += MP5990_EIN_SAMPLE_CNT_MAX;
+	}
+
+	energy_diff = (double)(rollover - pre_rollover) * MP5990_EIN_ENERGY_CNT_MAX +
+		      (double)energy - (double)pre_energy;
+	if (energy_diff < 0) {
+		LOG_DBG("Energy difference is less than zero.\n");
```
Comment (GoldenBug) [inline common/dev/mp5990.c:86]: remove newline

File: common/dev/mp5990.c [hunk 5]
```diff
@@ -24,6 +25,79 @@
...
+		return -1;
+	}
+
+	sample_diff = sample - pre_sample;
+	if (sample_diff == 0) {
+		LOG_DBG("Sample difference is less than zero.\n");
```
Comment (GoldenBug) [inline common/dev/mp5990.c:92]: remove newline.

**Review by GoldenBug — CHANGES_REQUESTED** [review]

**Comment by YoungLi123** [comment]
Hi @GoldenBug 
Thanks for suggestions. I pushed a commit. Please review it again.

**Pull request closed without merging**